battle.go

```diff
@@ -155,18 +151,12 @@ func (b *Battle) SimulateRound() ([]Transaction, bool) {
 			queue = queue[1:]
 			switch t := next.(type) {
 			case DamageTransaction:
-				if t.Target.CurrentHP >= t.Damage {
-					t.Target.CurrentHP -= t.Damage
-				} else {
-					// prevent underflow
-					t.Target.CurrentHP = 0
-				}
-				if t.Target.CurrentHP == 0 {
+				receiver := b.getPokemon(t.Target.party, t.Target.partySlot)
+				receiver.Damage(t.Damage)
```

Comment:
Not a huge fan of this, tbh. Particularly because the code it's replacing is extremely trivial, and because Damage is publicly facing.

Reply:
We can make things like heal, damage, cure, etc. be internal functions. The goal here is to make testing damage separate from testing damage transactions.

Reply:
Then we should separate the creation of transactions from the processing of transactions. All changes to the battle state should go through transactions anyway.

Here, we're just making the logic harder to read by requiring people to look in multiple places for it.